api/v1/controllers/collectors.js

```diff
@@ -270,12 +271,16 @@ function heartbeat(req, res, next) {
       throw new apiErrors.ForbiddenError({
         explanation: 'Authentication Failed',
       });
-    } else if (o.status !== 'Running' && o.status !== 'Paused' && o.status !== 'MissedHeartbeat') {
+    } else if (o.status === status.Stopped) {
       throw new apiErrors.ForbiddenError({
-        explanation: `Collector must be running or paused or missed. Status: ${o.status}`,
+        explanation: `Collector cannot send a heartbeat if stopped. Status: ${o.status}`,
       });
     }
 
+    if (o.status !== status.Running) {
+      o.set('status', status.Running);
+    }
+
     retval.collectorConfig.status = o.status;
```

Comment:
What's the purpose of sending back the status in the heartbeat? As it is right now it will always be "Running". Was it supposed to be that a collector can be marked as "Stopped" via the api and will be instructed to stop next time it sends a heartbeat?

Reply:
Correct. Gives a collector admin the ability to stop or resume a running or paused collector without needing access to the host where it's running.

Reply:
But if you stop a running collector via the api, it's status in the db will be set to "Stopped", and the next heartbeat it sends will be rejected, so the collector will never find out it was supposed to be stop.

@iamigo

Reply:
Oops -- loophole! Need to figure out how to allow a user to use API to stop a collector remotely. Pls. create story.

Reply:
will do.